src/aurelian/agents/knowledge_agent/knowledge_agent_tools.py

```diff
@@ -0,0 +1,36 @@
+"""Tools for the Knowledge Agent."""
+
+from typing import List, Tuple
+
+from oaklib import get_adapter
+
+
+async def search_ontology_with_oak(term: str, ontology: str, n: int = 10, verbose: bool = False) -> List[Tuple[str, str]]:
+    """
+    Search an OBO ontology for a term.
+
+    Note that search should take into account synonyms, but synonyms may be incomplete,
+    so if you cannot find a concept of interest, try searching using related or synonymous
+    terms.
+
+    If you are searching for a composite term, try searching on the sub-terms to get a sense
+    of the terminology used in the ontology.
+
+    Args:
+        term: The term to search for.
+        ontology: The ontology ID to search
+        n: The maximum number of results to return.
+        verbose: Whether to print debug information.
+
+    Returns:
+        A list of tuples, each containing an ontology ID and a label.
```

Comment:
fwiw, as a curator, i like to know whether the match is to the label or a synonym. also is the `n` the number of search results to return per search term?

Reply:
Thanks @twhetzel - I think probably the way to handle that is to preserve the span / original text so that the user can decide whether it's an exactly match, synonym, misspelling, etc. 

n here is the maximum number of matching terms to return